### DISCUSSION TITLE
Why does the wpm graph look like it's averaged throughout the game?

### DISCUSSION THREAD

**Discussion by ghost** [monkeytypegame/monkeytype#963] [Ideas] [answered]
I posted this in Ideas, but is not an idea, but there is only one category.

The wpm chart that you receive after you finish the test looks like it is averaged since the start of the game at every point, quite opposite to an instantaneous speed chart that shows your rough speed at every interval. I find this quite ironic, because I expected it to be an instantaneous speed graph. I don't notice this with the raw speed though, which looks like it's roughly instantaneous

I know that because, if I pause for a long while, then continue on with a test it looks like my speed has received a stunt for the rest of the game. The raw speed is much more meaningful in this scenario.
![index](https://user-images.githubusercontent.com/55037535/108131189-07871300-7105-11eb-8a09-4c247594e453.png)

The bump in the raw speed at the beginning is because you need to type something to start the round. If you look at the yellow wpm graph, it seems to be rising to the final wpm, which is clearly lower than what would be expected in this interval, since I was AFK most of the time at the beginning. Here is a two minute one.

![index](https://user-images.githubusercontent.com/55037535/108132020-49fd1f80-7106-11eb-947c-7c38de93ea9b.png)

If it were instantaneous, it would have been around the raw speed, because after I started typing, I clearly wasn't rising from 0 to 28 in a full minute .

**Comment by ghost** [comment, marked as answer]
It seems to be like this intentionally.

**Reply by Miodec** [reply]
Yeap - its by design
